docs/logs.md

```diff
@@ -0,0 +1,67 @@
+```
+
+For example:
+
+```
+2023-03-07T11:32:29.948Z WARN pkg_managers.pip No setup.py found in directory /home/fepas/cachi2/workdir/cachito-pip-with-deps, package is likely not pip compatible
+2023-03-07T11:32:29.948Z INFO pkg_managers.pip Filling in missing metadata from setup.cfg
+2023-03-07T11:32:29.949Z INFO pkg_managers.pip Found metadata {"name": "cachito-pip-with-deps", "version": "1.0.0" }
+2023-03-07T11:32:29.949Z INFO pkg_managers.pip Resolved package {"name": "cachito-pip-with-deps", "version": "1.0.0"}
+2023-03-07T11:32:29.951Z INFO pkg_managers.pip No hash options used, will not require hashes unless HTTP(S) dependencies are present.
+2023-03-07T11:32:29.959Z INFO pkg_managers.pip Dependencies found [{"name": "ruamel-yaml-clib", "version": "0.2.6"}, {"name": "six", "version": "1.16.0"}, {"name": "urllib3", "version": "1.26.7"}, {"name": "cython", "version": "0.29.33"}]
+2023-03-07T11:32:29.161Z INFO pkg_managers.pip Downloading dependencies [{"name": "ruamel-yaml-clib", "version": "0.2.6"}, {"name": "six", "version": "1.16.0"}, {"name": "urllib3", "version": "1.26.7"}, {"name": "cython", "version": "0.29.33"}]
+2023-03-07T11:32:29.161Z INFO pkg_managers.pip Dependencies downloaded successfully [{"name": "ruamel-yaml-clib", "version": "0.2.6"}, {"name": "six", "version": "1.16.0"}, {"name": "urllib3", "version": "1.26.7"}, {"name": "cython", "version": "0.29.33"}]
+2023-03-07T11:32:29.161Z INFO core.general The processing of the packages and dependencies has completed successfully {"output": "deps/pip/"}
+```
+
+The meaning of **timestamp** and log level should be self evident. The **logger** name helps understanding which part of Cachi2 is emitting the log. The **message** is a human readable string describing the event being logged. The **json data** contains additional fields useful for searching.
+
+Care should be taken to make sure specific key/value pairs are logged.
+
+### 1. When did it happen?
+
+**Included in:** `TIMESTAMP`
+
+Encode timestamps in UTC/ISO-8601 format at the start of the log line.
+
+### 2. What happened?
+
+**Included in:** `MESSAGE`, `ADDITIONAL_JSON_DATA`
+
+This should appear as a key in the `ADDITIONAL_JSON_DATA` at the end of the log line and can
+optionally appear in the human-readable `MESSAGE`.
+
+### 3. Where did it happen?
+
+**Included in:** `LOGGER`
+
+This should indicate which part of Cachi2 is emitting the log.
+
+## Consequences
+
+* By using structured logs that are only partially formatted as JSON, we should strike a balance
+  between easy readability and support for centralized queries. It should be easy for a human to
+  read Cachi2 logs directly since the first portion of each line contains a human
+  readable string early on, while the JSON formatted suffix supports the creation of queries that
+  span subsystems in centralized aggregated logs.
+
+
+Note: This document is based on https://github.com/redhat-appstudio/book/blob/main/ADR/0006-log-conventions.md.
```

Comment:
Similarly to the appstudio ADR, should we look into what this means from an implementation perspective?

I wouldn't want to log like this all over the codebase
```python
log.info("Dependencies found %s", json.dumps(dependencies))
```
It's probably OK to do the detailed design in a follow-up story, basically I'm just asking if structured logging is a thing in Python